File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        vector<int> sortedNums(nums.begin(), nums.end());
        sort(sortedNums.begin(), sortedNums.end());

        for (int i = 0; i < n; ++i){
            bool isMatch = true;
            for (int j = 0; j < n; ++j){
                if (sortedNums[(i + j) % n] != nums[j]){
                    isMatch = false;
                    break;
                }
            }
            if (isMatch) return true;
        }
        return false;
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        sorted_nums, n = nums[:], len(nums)
        sorted_nums.sort()
        
        for i in range(n):
            matching = True
            for j in range(n):
                if sorted_nums[(i + j) % n] != nums[j]:
                    matching = False
                    break

            if matching:
                return True 

        return False